### ISSUE TITLE
Search - Improve focus movement after interaction with nav elements

### ISSUE THREAD

**Issue by johnatawnclementawn** [archesproject/arches#9667]
Focus should be shifted after interacting with elements that show or hide content in Search. e.g. When a user clicks `Details` link, focus should shift to the report expander (first interactable element within the Details panel). Currently, a user needs to tab through the entire list of search-results. 

![Image](https://github.com/archesproject/arches/assets/22161487/fe42064a-5295-43f4-8347-a1fb169371fb)

Additionally, after interacting with the search toolbar popup items (time filter, saved search, search export), focus should shift into the opened panel, rather than tabbing through the Active search Tab

![Image](https://github.com/archesproject/arches/assets/22161487/597b43aa-e8e9-44ae-a6fd-a50db03567ef)

**Comment by johnatawnclementawn** [comment]
Once #9488 is merged, this comment will need to be addressed in this PR: https://github.com/archesproject/arches/pull/9488#pullrequestreview-1495756724

**Comment by johnatawnclementawn** [comment]
Resolved by #9723
